juice/src/net/container/sequential.rs

```diff
@@ -0,0 +1,317 @@
+    }
+
+    pub fn map_input(&mut self, name: &str) {
+        self.inputs.push(name.to_string());
+    }
+
+    pub fn map_output(&mut self, name: &str) {
+        self.outputs.push(name.to_string());
+    }
+}
+
+impl<B: IBackend + LayerOps<f32> + 'static> Sequential<B> {
+    pub fn new(mut descriptor: Descriptor, config: &SequentialConfig) -> Self {
+        // Create internal layers one by one and connect them.
+        // For the purpose of connecting layers, all inputs and outputs have names,
+        // which are either explicitly given in the config, or have implicit form of
+        // io_{i}_{j} for layer inputs and io_{i+1}_{j} for outputs, where i is the layer index
+        // (0-based) and j is the input/output index within the layer.
+        // io_0_{j} are inputs to the Sequential layer itself.
+
+        // All children layer output junction known so far, keyed by their internal names.
+        // This is initialized with Sequential inputs below.
+        let mut internal_junctions = HashMap::new();
+
+        // Output names from the previous layer. These will be assumed to be the inputs of the next
+        // layer unless it has explicit input names in the config.
+        // Initialized with Sequential input names below.
+        let mut prev_layer_output_names = Vec::new();
+
+        // Create an array of Sequential inputs where:
+        // * internal names are either explicitly set in config or implicitly set to "io_0_{j}",
+        // * shapes and data paths are taken from the descriptor.
+        for (j, input) in descriptor.inputs().iter().enumerate() {
+            let internal_name = if config.inputs.len() > j {
+                config.inputs[j].clone()
+            } else {
+                format!("io_0_{}", j)
+            };
+            prev_layer_output_names.push(internal_name.clone());
+            internal_junctions.insert(internal_name, input.junction.clone());
+        }
+
+        // Create children layers.
+        let mut children = Vec::new();
+        for (i, child_config) in config.layers.iter().enumerate() {
+            // Inputs for a child are either given explicitly in the config, or replicate the
+            // outputs of the previous child. Note that currently this doesn't support partial
+            // name specification (if explicit names are given, they are assumed to be the only
+            // inputs this layer has).
+            let child_input_names = if !child_config.inputs.is_empty() {
+                child_config.inputs.clone()
+            } else {
+                prev_layer_output_names
+            };
+
+            let child_inputs = child_input_names
+                .iter()
+                .map(|name| {
+                    Inout::new_with_junction(
+                        internal_junctions
+                            .get(name)
+                            .unwrap_or_else(|| panic!("Unknown input/output name {}", name))
+                            .clone(),
```

Comment:
`.cloned()` and a proper error would be nice

Reply:
Changed net creation functions to return `Result`.